make-release.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+# Release process automation script.
+# Used to create branch/tag and update VERSION files
+# and and trigger release by force pushing changes to the release branch
+
+# set to 1 to actually trigger changes in the release branch
+TRIGGER_RELEASE=0
+NOCOMMIT=0
+MAJOR_VERSION_BUMP=0
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [OPTIONS]"
+  echo "Options:"
+  echo "    --help, -h"
+  echo "        Show this message."
+  echo "    --no-commit, -n"
+  echo "        Do not commit changes."
+  echo "    --trigger-release, -t"
+  echo "        Push changes to remote repo, triggering release."
+  echo "Example: $0 --repo git@github.com:eclipse/che-subproject --version 'v3.2.1' --trigger-release"; echo
+}
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-r'|'--repo') REPO="$2"; shift 1;;
+    '-v'|'--version') VERSION="$2"; shift 1;;
+    '-t'|'--trigger-release')
+      if [[ ! ${NOCOMMIT} -eq 0 ]]; then
+        echo "Only one of --trigger-release and --no-commit may be specified"
+        exit 1
+      fi
+      TRIGGER_RELEASE=1; shift 0;;
+    '-n'|'--no-commit')
+      if [[ ! ${NOCOMMIT} -eq 0 ]]; then
+        echo "Only one of --trigger-release and --no-commit may be specified"
+        exit 1
+      fi
+      NOCOMMIT=1; shift 0;;
+    '-h'|'--help'|*) usage; exit 0;;
+  esac
+  shift 1
+done
+
+if [[ ! ${VERSION} ]] || [[ ! ${REPO} ]]; then
+  usage
+  exit 1
+fi
+
+if [[ ${VERSION} != "v"* ]]; then
+  echo "Version must be prefixed with 'v'"
+  exit 1
+fi
+
+CURRENT_MAJOR_VERSION=$(cut -f 1 -d '.' VERSION)
+NEW_MAJOR_VERSION=$(echo "$VERSION" | cut -f 1 -d '.')
+if [[ "${NEW_MAJOR_VERSION}" != "${CURRENT_MAJOR_VERSION}" ]]; then
+  MAJOR_VERSION_BUMP=1
+fi
+
+# derive branch from version
+BRANCH=${VERSION%.*}.x
+
+# if doing a .0 release, use master; if doing a .z release, use $BRANCH
+if [[ ${VERSION} == *".0" ]]; then
+  BASEBRANCH="master"
+else
+  BASEBRANCH="${BRANCH}"
+fi
+
+# work in tmp dir
+TMP=$(mktemp -d); pushd "$TMP" > /dev/null || exit 1
+
+# get sources from ${BASEBRANCH} branch
+echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+git clone "${REPO}" -q
+cd "${REPO##*/}" || exit 1
```

Comment:
```suggestion
cd ./* || exit 1
```
It should help to solve an issue when user copied/pasted URL from GitHub when ` git` appears in the end, like me
```
❯ ./make-release.sh --repo git@github.com:sleshchenko/che-plugin-broker.git --version v3.3.0 --no-commit
Check out git@github.com:sleshchenko/che-plugin-broker.git to /tmp/tmp.jNL8XGtytZ/che-plugin-broker.git
./make-release.sh: line 77: cd: che-plugin-broker.git: No such file or directory

❯ cd /tmp/tmp.jNL8XGtytZ/che-plugin-broker.git
cd: no such file or directory: /tmp/tmp.jNL8XGtytZ/che-plugin-broker.git

❯ cd /tmp/tmp.jNL8XGtytZ/che-plugin-broker/
# Success!!!!
```

Reply:
Makes sense to me.